static/js/components/forms/SiteAddForm.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react"
+import { ErrorMessage, Field, Form, Formik, FormikHelpers } from "formik"
+import * as yup from "yup"
+
+import { ConfigField, ConfigItem } from "../../types/websites"
+import { componentFromWidget } from "../../lib/util"
+
+export const websiteValidation = yup.object().shape({
+  title: yup
+    .string()
+    .label("Title")
+    .trim()
+    .required()
+})
+
+const renderField = (field: ConfigField) => (
+  <div key={field.name} className="form-group">
+    <label htmlFor={field.name} className="font-weight-bold">
+      {field.label}
+    </label>
+    <Field
+      as={componentFromWidget(field)}
+      name={field.name}
+      className="form-control"
+    />
+    <ErrorMessage name={field.name} component="div" />
+  </div>
+)
```

Comment:
You have some very similar code in `SiteEditForm.tsx` below - could be helpful to pull this out as a separate `SiteContentField` component that just takes a single prop of type `ConfigField`